src/fandango/evolution/algorithm.py

```diff
@@ -89,7 +92,10 @@ def __init__(
         self.tournament_size = tournament_size
         self.warnings_are_errors = warnings_are_errors
         self.best_effort = best_effort
+        self.current_max_nodes = max_nodes
+        self.diversity_k = diversity_k
```

Comment:
We shouldn't have information redundancy such as this. `AdaptiveTuner` owns this information, and it should be the only source of truth for this. If you need to be able to reset it, make `AdaptiveTuner` be able to do this. And then access the information from there directly to prevent out-of-sync errors.